server/schemas/schemaMatcher.js

```diff
@@ -0,0 +1,80 @@
+var csv = require("csv");
+var Readable = require("stream").Readable;
+var schemaModel = require("./schemaModel");
+var helpers = require("../config/helpers");
+
+var handle = function(buffer, callback) {
+  var stringifier = new csv.stringify();
+  var parser = new csv.parse();
+  var readStream = new Readable({objectMode: true});
+  var validEntry = true;
+  var storage = {};
+  var columns;
+
+//turns an array into a query which can be sent to mongoDB
+//via mongoose in order to find a schema template with the same fields.
+  var queryBuilder = function(array) {
+    var obj = {};
+    for (var i = 0; i < array.length; i++) {
+      //true is a placeholder value.
+      obj[array[i].toString()] = true;
+    }
+    //if the object is empty, the query would return all schemas. Therefore, we intercept this case
+    //and return false instead.
+    if (Object.keys(obj).length === 0) {
+      return false;
+    }
+    var result = {};
+    for (var prop in obj) {
+      if (obj.hasOwnProperty(prop)) {
```

Comment:
This made more sense for me than using _.transform because transform held input>output object consistency (arrays yield arrays, objects yield objects). I need to go from array>object.